activerecord/lib/active_record/relation/finder_methods.rb

```diff
@@ -459,7 +459,9 @@ def find_some(ids)
       end
 
       if result.size == expected_size
-        result
+        return result if self.values[:order]
+        records_by_id = result.index_by(&:id)
+        ids.first(expected_size).collect { |id| records_by_id[id.to_i] }
```

Comment:
@miguelgraz How would it work when `limit_value` is present?

For example, if we pass a list of 5 ids but the limit is 2, I understand the query will retrieve only 2 records. Since we're passing the complete list of ids to the `where` condition, we don't know in advance which ids are associated with these 2 records. 

In other words, I think we can not deduce that `ids.first(expected_size)` are the corresponding ones when `expected_size` is less than `ids.size` (when `limit` is used in the relation).